### PR TITLE
update ruff rules

### PR DESCRIPTION
**Pull request by ori-kron-wis**

### CODE REVIEW

File: tests/model/test_scvi.py [hunk 1]
```diff
@@ -506,7 +507,6 @@ def test_setting_adata_attr(n_latent: int = 5):
     # validation catches no batch
     with pytest.raises(KeyError):
         model.adata = adata3
-        model.get_latent_representation()
```
[the remark this answers is not in the document]
Comment (ori-kron-wis) [reply tests/model/test_scvi.py:509]: one of the new constraints does not allow more than 1 line under pytest raises error (only allow the line that casues the error which is the 1st line... so we never reach that 2nd one ).

File: pyproject.toml [hunk 2]
```diff
@@ -176,11 +176,16 @@ select = [
     "C4",  # flake8-comprehensions
     "BLE",  # flake8-blind-except
     "UP",  # pyupgrade
-    "RUF100",  # Report unused noqa directives
+    #"RUF100",  # Report unused noqa directives
+    #"RUF",  # Report unused noqa directives
```
Comment (canergen) [inline pyproject.toml:180]: I meant to activate this. Any issues with it? We should otherwise leave the RUF100 active.

File: pyproject.toml [hunk 3]
```diff
@@ -190,16 +195,17 @@ ignore = [
     # Errors from function calls in argument defaults. These are fine when the result is immutable.
     "B008",
     # __magic__ methods are are often self-explanatory, allow missing docstrings
-    "D105",
+    #"D105",
```
Comment (canergen) [inline pyproject.toml:198]: Should we uncomment again? It has no effect in the code

File: pyproject.toml [hunk 4]
```diff
@@ -190,16 +195,17 @@ ignore = [
...
     # first line should end with a period [Bug: doesn't work with single-line docstrings]
     "D400",
     # First line should be in imperative mood; try rephrasing
     "D401",
-    ## Disable one in each pair of mutually incompatible rules
-    # We don’t want a blank line before a class docstring
-    "D203",
```
Comment (canergen) [inline pyproject.toml:200]: Add this again? I don’t have preference here.

**Review by canergen — APPROVED** [review]
See comments. Otherwise LGTM.

File: tests/model/test_scvi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (canergen) [reply tests/model/test_scvi.py:509]: We should set adata and then run get_latent. So copy the first line out of the raise and add the second line.

[the remark this answers is not in the document]
Comment (canergen) [reply tests/model/test_scvi.py:509]: It’s at least my take of what we are testing here. Does setting model.adata fail? Then we should refine the comment on this test.

[the remark this answers is not in the document]
Comment (ori-kron-wis) [reply tests/model/test_scvi.py:509]: No, because model.adata = adata3 is what failing, We never reach the 2nd line.

File: pyproject.toml (hunk 4 above)
Comment (ori-kron-wis) [reply pyproject.toml:200]: Please suggest a fix on top and lets see tests finish again

File: tests/model/test_scvi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (canergen) [reply tests/model/test_scvi.py:509]: Got it thanks.

File: pyproject.toml (hunk 2 above)
Comment (canergen) [reply pyproject.toml:180]: Remove RUF line and activate RUF100 again.

**Pull request merged**